recipes/meson/all/test_package/conanfile.py

```diff
@@ -7,13 +7,15 @@ class TestPackageConan(ConanFile):
     generators = "pkg_config"
 
     def build(self):
-        meson = Meson(self)
-        meson.configure(build_folder="build")
-        meson.build()
+        if not tools.cross_building(self):
+            meson = Meson(self)
+            meson.configure(build_folder="build")
+            meson.build()
 
     def test(self):
-        bin_path = os.path.join("build", "test_package")
-        self.run(bin_path, run_environment=True)
+        if not tools.cross_building(self):
+            bin_path = os.path.join("build", "test_package")
+            self.run(bin_path, run_environment=True)
 
         self.run("meson --version")
```

Comment:
```suggestion
        self.run("meson --version", run_environment=True)
```

Reply:
This should not be needed, because meson recipe sets self.env_info.PATH. this is why the recipe passes

Reply:
The env path will not pass when using 2 profile builds.
e.g. this will fail:
```
conan create . meson/0.59.2@ -pr:h default -pr:b default
```

Reply:
indeed, on catalina the test log for meson/0.56.0 has
```
----Running------
> meson --version
-----------------
0.56.0
```
so it's clearly using system meson. thanks for noticing !